README.md

```diff
@@ -1,2 +1,279 @@
-# strapi-plugin-comments
-strapi-plugin-comments
+# Strapi - Comments plugin
+
+A plugin for [Strapi Headless CMS](https://github.com/strapi/strapi) that provides end to end comments feature with their moderation panel, bad words filtering, abuse reporting and more.
+
+### ⏳ Installation
+
+Install Strapi with this **Quickstart** command to create a Strapi project instantly:
+
+(Use **yarn** to install this plugin within your Strapi project (recommended). [Install yarn with these docs](https://yarnpkg.com/lang/en/docs/install/).)
+
+```bash
+yarn add strapi-plugin-comments@latest
+```
+
+
+Enjoy 🎉
+
+### 🖐 Requirements
+
+Complete installation requirements are exact same as for Strapi itself and can be found in the documentation under <a href="https://strapi.io/documentation/v3.x/installation/cli.html#step-1-make-sure-requirements-are-met">Installation Requirements</a>.
+
+**Supported Strapi versions**:
+
+- Strapi v3.0.5 (recently tested)
+- Strapi v3.x
+
+(This plugin may work with the older Strapi versions, but these are not tested nor officially supported at this time.)
+
+**We recommend always using the latest version of Strapi to start your new projects**.
+
+## Features
+
+- **Comments Public API:** Elegant, entirely customizable and a fully extensible admin panel.
+- **Strapi &amp generic users:** Support for built-in Strapi as well as for generic non-Strapi user as comments authors in a same time, depends on what type is users your UI is handling.
+- **Any Content Type relation:** Comments can by linked to any of your Content Types by default. Simply, you're controlling it.
+- **Moderation Panel:** Search & Filter through the bucket with your auditory comments. Manage them by blocking single ones or full threads. All in combined list &amp; hierarchical tree view of threads.
+- **Automated Bad Words filtering:** By detault end users are not allowed to post abusing comments where bad words has been used.
+- **Abuse Reporting & Reviewing:** Built on top of Node.js, Strapi delivers amazing performance.
+
+## Public API Comment model
+
+### Generic (non Strapi User)
+```
+{
+    "id": 1,
+    "content": "My comment content",
+    "blocked": null,
+    "blockedThread": true,
+    "blockReason": null,
+    "points": 1,
+    "authorUser": null,
+    "authorId": "PLE12345678",
+    "authorName": "Joe Doe",
+    "authorEmail": "jdoe@sample.com",
+    "authorAvatar": null,
+    "created_at": "2020-07-14T20:13:01.649Z",
+    "updated_at": "2020-07-14T20:13:01.670Z"
+}
+```
+### Strapi User
+```
+{
+    "id": 1,
+    "content": "My comment content",
+    "blocked": true,
+    "blockedThread": null,
+    "blockReason": null,
+    "points": null,
+    "authorUser": {
```

Comment:
Is it possible to change this into single user object? It is confusing to have `authorUser. username` and `authorName` and so on

if it can be either that type of user or the other then maybe user object with type would fit? (discriminated union in TS terms). Can we have both `authorUser` fields populated and the flat `author*` at the same time?

Reply:
Strapi flavor.